File: Python/strStr.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        if len(haystack) < len(needle):
            return -1
        mainString = len(haystack)
        subString = len(needle)
        i = 0
        while i < mainString - subString + 1:
            if haystack[i:i + subString] == needle:
                return i
            i = i + 1

        return -1


if __name__ == '__main__':
    solution = Solution()
    result = solution.strStr('hello', 'll')
    print(result)

File: src/com/example/helloworld/strStr.java
package com.example.helloworld;

class SolutionstrStr {
    public int strStr(String haystack, String needle) {

        if (needle == null) return 0;
        for (int i = 0; i < haystack.length() - needle.length() + 1; i++) {
            System.out.println(i);
            if (haystack.substring(i, i + needle.length()).equals(needle)) {
                return i;
            }
        }
        return -1;

    }
}

public class strStr {

    public static void main(String[] args) {

        SolutionstrStr solutionstrStr = new SolutionstrStr();
        solutionstrStr.strStr("hello", "ll");
    }
}
